evm/src/main/java/org/hyperledger/besu/evm/processor/AbstractMessageProcessor.java

```diff
@@ -158,7 +167,10 @@ private void completedSuccess(final MessageFrame frame) {
    *
    * @param frame The message frame
    */
-  private void completedFailed(final MessageFrame frame) {
+  private void completedFailed(final MessageFrame frame, final OperationTracer operationTracer) {
+    if (operationTracer != null) {
```

Comment:
I'd rather see these in the `process` method and these signatures not altered (like was done with the `CODE_SUSPENDED` case).  The `contextEnter` call also occurs there.

Reply:
In which scenario can the `contextEnter` happen here?